pkg/vendir/fetch/hg/hg.go

```diff
@@ -134,25 +139,25 @@ hgauth.password = %s
 			sshCmd = append(sshCmd, "-o", "StrictHostKeyChecking=no")
 		}
 
-		authRc = fmt.Sprintf("%s\n[ui]\nssh = %s\n", authRc, strings.Join(sshCmd, " "))
+		hgRc = fmt.Sprintf("%s\n[ui]\nssh = %s\n", hgRc, strings.Join(sshCmd, " "))
 	}
 
-	if (authOpts.Username != nil && authOpts.Password != nil) || authOpts.IsPresent() {
-		credsHgRcPath := filepath.Join(authDir, "hgrc")
-		err = ioutil.WriteFile(credsHgRcPath, []byte(authRc), 0600)
+	if (authOpts.Username != nil && authOpts.Password != nil) || authOpts.IsPresent() || len(t.opts.Hgrc) > 0 {
```

Comment:
this could be just simplified if hgRc is non-empty.

Reply:
fixed :heavy_check_mark: